ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -1657,17 +1633,17 @@ FEDataManager::interpWeighted(const int f_data_idx,
                     F_node, n_vars, X_node, NDIM, f_sc_data, patch, interp_box, interp_spec.kernel_fcn);
             }
 
-            // Insesrt the values of F at the nodes.
+            // Insert the values of F at the nodes.
             F_vec.insert(F_node, F_node_idxs);
         }
+        F_vec.close();
```

Comment:
We can completely remove all parallel communication in this code by utilizing the fact that we already have all the ghost data we need - the values of the lumped mass matrix don't change. If we work with the local PETSc versions (i.e., `VecGhostGetLocalForm()` and `VecGhostRestoreLocalForm()`) then we can scale things and get correct ghost values.

Reply:
So what we are doing here is to interpolate the the IB points that correspond to the given Cartesian grid data. Those IB points may be owned by another processor, and so I don't see how we can avoid communication.

That said, what this particular routine is doing is a bit perverse. With nodal interpolation, we do not have to do an L2 projection; we are interpolating the background field directly to the nodes of the FE mesh.

Here I am doing interpolating to the nodes and then _weighting_ the result using the diagonal mass matrix, so that the units are the same as in the other code path. Then, we go back and "solve" the L2 projection in `IBFEMethod` --- except that for nodal interpolation, it is the trivial projection using the diagonal mass matrix.

Reply:
(In other words, if we did not weight the result in `interWeighted()` using the diagonal mass matrix, the we also would not have to go back to do a subsequent solve.)